sx.el

```diff
@@ -259,6 +259,23 @@ whenever BODY evaluates to nil."
        :filter (lambda (&optional _)
                  (when (progn ,@body) ,def)))))
 
+(defmacro sx--create-comparator (name doc compare-func get-func)
+  "Define a new comparator called NAME with documentation DOC.
+COMPARE-FUNC is a function that takes the return value of
+GET-FUNC and performs the actual comparison."
+  (declare (indent 1) (doc-string 2))
+  (let ((gf (intern (format " %S--get-prop-function" name)))
+        (cf (intern (format " %S--compare-function"  name))))
+    ;; Leading space to hide from completion systems
+    `(progn
+       ;; In using `defalias', the macro supports both function
+       ;; symbols and lambda expressions.
+       (defalias ',gf ,get-func)
+       (defalias ',cf ,compare-func)
+       (defun ,name (a b)
+         ,doc
+         (,cf (,gf a) (,gf b))))))
```

Comment:
Yes we can.

The following file:

```
;; -*- lexical-binding:t -*-

(defun ok ()
  ""
  ((lambda (x) (message "%s" x)) 12))
(defun ok2 ()
  ""
  (message "%s" 12))
```
Compiles to this:
```
#@8 
(fn)
(defalias 'ok #[0 "\300\301\302\"\207" [12 message "%s"] 4 (#$ . 409)])
#@8 

(fn)
(defalias 'ok2 #[0 "\300\301\302\"\207" [message "%s" 12] 3 (#$ . 494)])
```

Note that there was no leftover reference to `x`, meaning it got optimized away. However, I get a compiler warning: `Warning: Use of deprecated ((lambda (x) ...) ...) form`